sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -262,6 +293,35 @@ AccessToken TokenCredentialImpl::ParseToken(
   accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
   accessToken.ExpiresOn = std::chrono::system_clock::now();
 
+  // expiresIn = number of seconds until refresh.
+  // expiresOn = timestamp of refresh expressed as seconds since epoch.
+
+  if (!refreshInPropertyName.empty() && parsedJson.contains(refreshInPropertyName))
+  {
+    auto const& refreshIn = parsedJson[refreshInPropertyName];
+    if (refreshIn.is_number_unsigned())
+    {
+      try
+      {
+        // 'refresh_in' as number (seconds until refresh)
+        auto const value = refreshIn.get<std::int64_t>();
+        if (value <= MaxExpirationInSeconds)
+        {
+          static_assert(
```

Comment:
Why does this static assert appear in multiple places? Why not have it once, immediately after the MaxExpirationInSeconds is defined?

Reply:
The static assert is for the `value` coming from the json response (not for `MaxExpirationInSeconds`). It shows up for each of the JSON properties being parsed (`expires_in`, `expires_on`, `refresh_in`, etc.), hence why it shows up multiple times (for each value).